isovar/protein_sequence.py

```diff
@@ -41,31 +43,92 @@ class ProteinSequence(TranslationKey):
         "translations",
     ]
 
-    def __init__(self, translations):
+    def __init__(
+            self,
+            amino_acids,
+            contains_mutation,
+            mutation_start_idx,
+            mutation_end_idx,
+            ends_with_stop_codon,
+            frameshift,
+            translations):
         """
-        Initialize fields of ProteinSequence. Fields inherited from TranslationKey
-        (e.g. frameshift, ends_with_stop_codon, &c) are inferred from the
-        translation objects which must all have the same values for these
-        fields.
+        Parameters
+        ----------
+        amino_acids : str
+            Amino acid sequence
+
+        contains_mutation : bool
+            Does the amino acid sequence contain a mutation?
+
+        mutation_start_idx : int
+            Start of half-open interval for variant amino acids
+            in the translated sequence
+
+        mutation_end_idx : int
+            End of half-open interval for variant amino acids
+            in the translated sequence
+
+        ends_with_stop_codon : bool
+            Did the amino acid sequence end due to a stop codon or did we
+            just run out of sequence context around the variant?
+
+        frameshift : bool
+            Was the variant a frameshift relative to the reference sequence?
```

Comment:
why are these special-cased?

Reply:
Holdover from old Isovar code, didn't want to change too much at once. I don't think we should care anymore whether a mutant amino acid sequence came from a frameshift but getting rid of that seems like it should be a different PR.